packages/amplify_core/lib/src/types/models/model_association.dart

```diff
@@ -14,34 +14,49 @@
  */
 import 'dart:convert';
 
+import 'package:aws_common/aws_common.dart';
 import 'package:flutter/foundation.dart';
 
 // ignore: constant_identifier_names
 enum ModelAssociationEnum { HasMany, HasOne, BelongsTo }
 
 @immutable
-class ModelAssociation {
+class ModelAssociation with AWSEquatable<ModelAssociation>, AWSSerializable {
   final ModelAssociationEnum associationType;
-  final String? targetName; // opt (used in belongsTo)
-  final String? associatedName; // opt (used in hasMany/hasOne)
-  final String? associatedType; // opt (used in hasMany/hasOne)
+  final String? targetName; // used in belongsTo
+  final List<String>? targetNames; // used in belongsTo
+  final String? associatedName; // used in hasMany/hasOne
+  final String? associatedType; // used in hasMany/hasOne
 
   const ModelAssociation({
     required this.associationType,
-    this.targetName,
+    @Deprecated('Please use the latest version of Amplify CLI to regenerate models')
```

Comment:
Do want this `@Deprecated` message to state what they should use instead of `targetName`?